src/main/java/hudson/plugins/throttleconcurrents/pipeline/ThrottleStepExecution.java

```diff
@@ -0,0 +1,69 @@
+package hudson.plugins.throttleconcurrents.pipeline;
+
+import hudson.model.Computer;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.throttleconcurrents.ThrottleJobProperty;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.workflow.steps.BodyExecution;
+import org.jenkinsci.plugins.workflow.steps.BodyExecutionCallback;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+
+import javax.annotation.Nonnull;
+
+public class ThrottleStepExecution extends StepExecution {
+    private final ThrottleStep step;
+
+    private BodyExecution body;
+
+    public ThrottleStepExecution(@Nonnull ThrottleStep step, StepContext context) {
+        super(context);
+        this.step = step;
+    }
+
+    public String getCategory() {
+        return step.getCategory();
+    }
+
+    @Override
+    public boolean start() throws Exception {
+        ThrottledStepInfo info = new ThrottledStepInfo(getCategory());
+
+        ThrottledStepInfo parentInfo = getContext().get(ThrottledStepInfo.class);
+        if (parentInfo != null) {
+            // Make sure we record the node used for the parent throttle step if it exists.
+            Computer computer = getContext().get(Computer.class);
+            if (computer != null && computer.getNode() != null) {
```

Comment:
Return `Node` from `getRequiredContext` and skip both the null check and the `getNode` conversion.